packages/x-date-pickers/src/internals/hooks/useMobilePicker/useMobilePicker.tsx

```diff
@@ -76,9 +76,11 @@ export const useMobilePicker = <
       return {
         ...inputPropsPassedByField,
         ...externalInputProps,
-        disabled: props.disabled,
-        onClick: props.readOnly || props.disabled ? undefined : actions.onOpen,
-        onKeyDown: onSpaceOrEnter(actions.onOpen),
```

Comment:
This was the oversight still allowing interactivity in mobile pickers with `readOnly` flag.